lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1162,33 +1172,40 @@ class MongoClientInterface {
         const masterKey = formatMasterKey(objName, params.vFormat);
         const versionKey = formatVersionKey(objName, params.versionId, params.vFormat);
         const _vid = generateVersionId(this.replicationGroupId);
-        const mst = generatePHDVersion(_vid);
-        c.bulkWrite([{
-            updateOne: {
-                filter: {
+        async.series([
+            next => c.updateOne(
+                {
                     _id: masterKey,
                 },
-                update: {
-                    $set: { _id: masterKey, value: mst },
-                },
-                upsert: true,
-            },
-        }, {
-            deleteOne: {
-                filter: {
-                    _id: versionKey,
+                {
+                    $set: {
```

Comment:
wondering if we should have a condition here, e.g. update the master conditionally when value is the indeed the versionId we are removing ? If the condition fails, we could then just fallback to deleting the object without PHD, as another version has been pushed already...

(not really needed for this specific bug fix, but may avoid setting a PHD when the version pointing to master is removed at the same time as a new version is added)

Reply:
Yes i think we should do this in a separate improvement ticket

Reply:
ticket: ARSN-275